timemachine/fe/single_topology.py

```diff
@@ -1047,12 +1046,12 @@ def assert_torsions_defined_over_non_linear_angles(system: GuestSystem | HostGue
     """
     linear_angles = set()
 
-    for idxs, angle_params in zip(system.angle.potential.idxs, system.angle.params):
+    for (i, j, k), angle_params in zip(system.angle.potential.idxs, system.angle.params):
         angle_k, angle_a0 = angle_params[0], angle_params[1]
 
         if angle_k > 0:
             if abs(angle_a0 - np.pi) < 0.174533:  # 10 degrees, arbitrary but conservative threshold
-                linear_angles.add(tuple(idxs))
+                linear_angles.add(tuple((i, j, k)))
```

Comment:
nit: redundant `tuple`?
```suggestion
                linear_angles.add((i, j, k))
```

Reply:
I tried that initially, but mypy really didn't like it:

```
         if angle_k > 0:
             if abs(angle_a0 - np.pi) < 0.174533:  # 10 degrees, arbitrary but conservative threshold
-                linear_angles.add(tuple((i, j, k)))
+                linear_angles.add((i, j, k))
```

Results in:

```
mypy.....................................................................Failed
- hook id: mypy
- exit code: 1

timemachine/fe/single_topology.py:1058: error: Unsupported operand types for in ("tuple[int, ...]" and "set[tuple[Any, Any, Any]]")  [operator]
timemachine/fe/single_topology.py:1062: error: Unsupported operand types for in ("tuple[int, ...]" and "set[tuple[Any, Any, Any]]")  [operator]
timemachine/fe/single_topology.py:1069: error: Unsupported operand types for in ("tuple[int, ...]" and "set[tuple[Any, Any, Any]]")  [operator]
timemachine/fe/single_topology.py:1073: error: Unsupported operand types for in ("tuple[int, ...]" and "set[tuple[Any, Any, Any]]")  [operator]
Found 4 errors in 1 file (checked 1 source file)
````